### PR TITLE
CMS 5 compatibility

### PR DESCRIPTION
**Pull request by GuySartorelli**
Adds CMS 5 compatibility - works with the beta, and with the change freeze on API breaking changes it should work with stable when that's released too.

It'll probably pay to write some unit or behat tests for this (there's some nifty email helper stuff with behat tests where you can check if emails were sent, to whom, what they contained, etc) but since there weren't tests before I won't be adding them in this PR.

### CODE REVIEW

File: code/email/MailblockMailSubscriber.php [hunk 1]
```diff
@@ -94,29 +99,5 @@ public function beforeSendPerformed(\Swift_Events_SendEvent $evt)
             $message->setBcc($bcc);
             $message->setCc($cc);
         }
-        else {
```
Comment (ChloeSartorelli) [inline code/email/MailblockMailSubscriber.php:97]: Everything from this else statement down is just double handling stuff that core already does. If you find these don't work out of the box, please raise an issue in github and I'll take a gander.

File: code/email/MailblockMailSubscriber.php [hunk 2]
```diff
@@ -2,65 +2,70 @@
...
+            MessageEvent::class => 'onMessage',
+        ];
+    }
+
+    public function onMessage(MessageEvent $event): void
     {
-        /** @var \Swift_Message $message */
-        $message = $evt->getMessage();
+        /** @var Email $message */
+        $message = $event->getMessage();
```
Comment (ChloeSartorelli) [inline code/email/MailblockMailSubscriber.php:27]: Technically it's a `Symfony\Component\Mime\RawMessage` which `Email` is now a subclass of - but the main mail event subscriber will already throw an exception if it's not specifically an `Email` instance anyway so we don't have to check that here.

**Comment by sig-steve** [comment]
Hi @GuySartorelli, sorry for the long delay in looking at this.

If I try to send a test e-mail I get a "server error" popup. The network response seems to be indicating an invalid argument.

[Emergency] Uncaught TypeError: SilverStripe\Control\Email\Email::__construct(): Argument #5 ($cc) must be of type array|string, null given, called in /home/webspace/ss5/site/vendor/signify-nz/silverstripe-mailblock/code/extensions/MailblockSiteConfigLeftAndMain.php on line 40
POST /admin/settings/EditForm

Line 164 in /home/webspace/ss5/site/vendor/silverstripe/framework/src/Control/Email/Email.php
Source

```
155                     $encoded .= '&#x' . bin2hex($email[$x]) . ';';
156                 }
157                 return $encoded;
158             default:
159                 user_error('Email::obfuscate(): Unknown obfuscation method', E_USER_NOTICE);
160                 return $email;
161         }
162     }
163 
164     public function __construct(
165         string|array $from = '',
166         string|array $to = '',
167         string $subject = '',
168         string $body = '',
169         string|array $cc = '',
170         string|array $bcc = '',
```

**Comment by sig-steve** [comment]
I can add the stack trace if you think it'll be useful.

**Comment by ChloeSartorelli** [comment]
Oops sorry about that. The values from those form fields must be coming back null instead of an empty string if they're empty. Fixed now.
I also noticed that the filename and class name for `MailblockMailSubscriber` didn't match in my original PR so I've fixed that discrepency as well.

Did a quick test with various combinations of fields empty - looks to be working as expected now.

**Comment by sig-lani** [comment]
I can confirm, this compatibility fix works (tested). Recommend merging and releasing new version branch.

**Pull request merged**